File: examples/js/get-scans.js
/**
 * QrVeloz REST API — List QR codes sorted by scan count (highest first)
 * Requires Node.js 18+ (uses native fetch)
 *
 * The list endpoint sorts by total scan count but does NOT return the count
 * per code in the response body. To get the exact scan count for a specific
 * code, use the MCP get_qr_scans tool or call GET /api/v1/qr-codes/{id}
 * and check the QrVeloz dashboard for full analytics.
 *
 * Usage:
 *   QRVELOZ_API_KEY=qrv_... node get-scans.js
 */

const API_BASE = 'https://qrveloz.com/api/v1';
const API_KEY  = process.env.QRVELOZ_API_KEY;

if (!API_KEY) {
  console.error('Error: set QRVELOZ_API_KEY environment variable');
  process.exit(1);
}

async function listQrCodes() {
  const response = await fetch(`${API_BASE}/qr-codes?limit=100&sort=scanCount&order=desc`, {
    headers: { 'Authorization': `Bearer ${API_KEY}` },
  });

  if (!response.ok) {
    const error = await response.json().catch(() => ({ error: response.statusText }));
    throw new Error(`${response.status}: ${error.error}`);
  }

  return response.json();
}

const { items, total } = await listQrCodes();

console.log(`QR codes sorted by scan count (${total} total):\n`);

for (const qr of items) {
  const status = qr.isActive ? 'active' : 'inactive';
  console.log(`  ${qr.title.padEnd(30)}  ${qr.shortCode}  →  ${qr.targetUrl}  [${status}]`);
}

console.log('');
console.log('Note: scan counts are not included in the list response.');
console.log('Use the MCP get_qr_scans tool for per-code scan totals.');

File: examples/python/get_scans.py
"""
QrVeloz REST API — Get scan counts for all QR codes
Requires Python 3.8+ and the requests library: pip install requests

Usage:
    QRVELOZ_API_KEY=qrv_... python get_scans.py
"""

import os
import sys
import requests

API_BASE = "https://qrveloz.com/api/v1"
API_KEY  = os.environ.get("QRVELOZ_API_KEY")

if not API_KEY:
    print("Error: set QRVELOZ_API_KEY environment variable")
    sys.exit(1)


def list_qr_codes_by_scans() -> dict:
    response = requests.get(
        f"{API_BASE}/qr-codes",
        headers={"Authorization": f"Bearer {API_KEY}"},
        params={"limit": 100, "sort": "scanCount", "order": "desc"},
    )
    response.raise_for_status()
    return response.json()


if __name__ == "__main__":
    data  = list_qr_codes_by_scans()
    items = data["items"]
    total = data["total"]

    print(f"Scan summary ({total} QR codes, sorted by scan count):\n")

    for qr in items:
        scan_count = qr.get("scanCount") or 0
        bar = "█" * min(scan_count, 40)
        print(f"  {qr['title'][:30]:<30}  {scan_count:>6} scans  {bar}")
